src/PerfView/PerfViewData.cs

```diff
@@ -5744,6 +5764,27 @@ protected internal override StackSource OpenStackSourceImpl(string streamName, T
                                     }
                                     sumCumMetric += cumMetric;
                                     cumCount++;
+
+                                    string GetAllocationType()
```

Comment:
I don't actually like local functions that get too much of their state implicitly via variable capture).   In particular at least pass things that represent non-contextual state (in this case the callerIndex).